src/addBusinessDays/index.ts

```diff
@@ -42,49 +42,64 @@ export default function addBusinessDays(
       ? [1, 2, 3, 4, 5]
       : options.businessDays.filter((number) => number < 7).map(toInteger)
 
-  const date = toDate(dirtyDate)
+  const initialDate = toDate(dirtyDate)
   const amount = toInteger(dirtyAmount)
   let accountedForExceptions = 0
+
   const isExcepted = (date: Date): boolean | null => {
     if (options.exceptions) {
       const exception =
         Object.keys(options.exceptions).find((e) =>
           isSameDay(new Date(e), date)
         ) || ''
-      if (options.exceptions[exception] !== undefined) {
+
+      if (options.exceptions[exception] === true) {
+        accountedForExceptions -= 1
+      } else if (options.exceptions[exception] === false) {
         accountedForExceptions += 1
       }
       return options.exceptions[exception]
     }
     return null
   }
-  const isNonWorkingDay = (date: Date) =>
-    isExcepted(date) === false || !businessDays.includes(date.getDay())
+
+  const isWorkingDay = (date: Date) => {
+    const hasException = isExcepted(date)
+    if (hasException === false) return false
+    if (hasException === true || businessDays.includes(date.getDay())) {
```

Comment:
previously, if the exception was true, this function would return false because it wasn't a business day

Reply:
In one of the cases that Khalid fixed the other day, the behaviour of using `sign` as is resulted in too few business days being added when starting on a non-working day. This switch got it working, but at the expense of your new cases 🤔